### PR TITLE
feat(containers): Added container queries

### PR DESCRIPTION
**Pull request by carriemorrison**
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

---

### Description

Added support for container queries, using the same breakpoint logic since that is vague in description and allows users to config the required string. It is possible to use NON named containers though I don't think that is very useful for atomic css. Using named containers relies on adding extra config as described in the updated documentation.

### Motivation and Context

Because Container queries! but Seriously very helpful to be able to use them.

### How Has This Been Tested?

Added an example file using named containers.

### Types of changes

What types of changes does your code introduce? Put an x in all the boxes that apply:

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

Review the following points, and put an x in all the boxes that apply. If you’re unsure about any of these, don’t hesitate to ask. We’re here to help!

-   [x] I have updated the documentation accordingly.
-   [x] I have read the [CONTRIBUTING](https://github.com/acss-io/atomizer/blob/main/CONTRIBUTING.md) document.
-   [x] I have added tests to cover my changes.

*reference*

Screenshot 2023-03-31 at 2 40 18 PM

*Example gif*

container

*Documentation*
0 0 0 0_4000_breakpoints html

### CODE REVIEW

File: docs/breakpoints.md [hunk 1]
```diff
@@ -97,3 +97,84 @@ When using explicit breakpoints, use multiple classes to have styles applied in
...
+We have added support for container queries using the same syntax as media queries, since you are able to define what the rule is, just ensure the names are different to the media query rules.
+
+```js
+breakPoints: {
+    sm: '@media screen and (min-width: 700px)',
+    mw300: '@container (max-width: 300px)',
+},
+```
+
+Same logic as above - append `--<breakpoint name>` to any Atomic class to associate that styling with the breakpoint of your choice. For example, `Fz(1rem)--mw300` will create the following rule in the related container query:
```
Comment (redonkulus) [inline docs/breakpoints.md:113]:
```suggestion
Same logic as above - append `--<breakpoint name>` to any Atomizer class to associate that styling with the breakpoint of your choice. For example, `Fz(1rem)--mw300` will create the following rule in the related container query:
```

File: docs/breakpoints.md [hunk 2]
```diff
@@ -97,3 +97,84 @@ When using explicit breakpoints, use multiple classes to have styles applied in
...
+
+```css
+@container width300 (max-width: 300px) {
+  .Fz(1rem)--contmw300 {
+    font-size: 1rem;
+  }
+}
+```
+
+```html
```
Comment (redonkulus) [inline docs/breakpoints.md:155]: Would be good to add a sentence above just to differentiate this example from the one above.

File: .changeset/tall-lions-give.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"atomizer": minor
+---
+
+Added container queries
```
Comment (redonkulus) [inline .changeset/tall-lions-give.md:5]:
```suggestion
feat: Added container queries support
```

File: packages/atomizer/src/rules.js [hunk 4]
```diff
@@ -1343,6 +1343,38 @@ module.exports = [
     },
     /**
     ==================================================================
+    CONTAINERS
+    ==================================================================
+    // TODO: Discuss `n` , `nor`, `z`
```
Comment (redonkulus) [inline packages/atomizer/src/rules.js:1348]: you can remove this, what you have is good

File: packages/atomizer/src/rules.js [hunk 5]
```diff
@@ -1343,6 +1343,38 @@ module.exports = [
...
+        name: 'Container Type',
+        matcher: 'ContType',
+        allowParamToValue: false,
+        styles: {
+            'container-type': '$0',
+        },
+        arguments: [
+            {
+                is: 'inline-size',
+                n: 'none',
```
Comment (carriemorrison) [inline packages/atomizer/src/rules.js:1370]: this follows existing pattern
```
                n: 'none',
                nor: 'normal',
```

File: packages/atomizer/src/rules.js [hunk 6]
```diff
@@ -1343,6 +1343,38 @@ module.exports = [
...
+                nor: 'normal',
+                s: 'size',
```
Comment (carriemorrison) [inline packages/atomizer/src/rules.js:1372]: we only have 1 other 'size' and that exists in same object as a `style`
```
            {
                n: 'none',
                st: 'strict',
                c: 'content',
                z: 'size',
                l: 'layout',
                s: 'style',
                p: 'paint',
            },
```

we could follow it and use `z` or keep `s` as we also have `is` for `inline-style`

**Review by redonkulus — APPROVED** [review]

**Pull request merged**